### ISSUE TITLE
Offer to write a simpler tutorial

### ISSUE THREAD

**Issue by Russ76** [ros-navigation/navigation2_auxiliary#1]
Steve: I appreciate your hard work on the ROS2 Nav2 stack! I hope and expect that Samsung assists with this role.

I have found some of the tutorial and explanation documents rather advanced. Often they sound like they are written by experts for experts. However, many of the readers are not experts! Many are young and some are just beginners in robotics. 

I would like to offer assistance in writing some tutorials aimed at a junior high grade level reader. These, in conjunction with the regular "official" guides, may assist hobbyists and readers and promote their successful completion of their robot projects.

Thanks again,
Russ

**Comment by SteveMacenski** [comment]
I'd potentially be interested in that! Did you have some suggested pages or roadmap for what you're be looking to add?

**Comment by Russ76** [comment]
Yes, one thing I'd start with is how to write a script to do waypoint following. Like, what details must be included, where to put this file, how to "compile" it and how to launch it. These details are hardly mentioned in the regular docs. Many of the parts needed for this are in the Nav2 files but they are scattered in different folders.

**Comment by SteveMacenski** [comment]
Hi,

We tend to shy away from talking about a user's application system since it can very widely and there are many different potential avenues that make sense for folks. I think something like that could definitely be useful information to post on a blog or elsewhere on the internet, but not something that we'd plan to have in the Nav2 docs without complete counter examples for the main categorizations of autonomy system structures to offer a relatively complete overview of options.  Else, folks may read this as if waypoint following scripts 'are the only' way to do it with Nav2

**Comment by Russ76** [comment]
Why doesn't Slam Toolbox install with Navigation 2? Isn't it required to do SLAM? Why is "backtrace" described before Slam toolbox is mentioned in the docs? Does a beginner need to know about backtrace and GDB and other popular debuggers? Is writing a behavior tree script required before one can use Nav2? I can picture people throwing up their hands wondering how to get started.

**Comment by SteveMacenski** [comment]
I believe we have a Getting Started guide that's the first thing on the website: https://navigation.ros.org/getting_started/index.html

The list of tutorials are in no particular order because there is no particular order for them. There's no evolution over time building on previous tutorials to gain a deeper understanding - these are all independent topics. I agree though that perhaps the most obvious ones (e.g. Navigating with a Physical Turtlebot 3, Navigating While Mapping, Using an External Costmap Plugin) should probably go first. I'll make that change shortly 

> Why doesn't Slam Toolbox install with Navigation 2?

Its another independent project from Nav2 and its pretty common for folks to swap out SLAM and localization solutions for what's appropriate for their applications.

**Comment by Russ76** [comment]
Thanks for the consideration, Steve, I know you're a busy guy!
